go/tasks/plugins/array/core/state.go

```diff
@@ -249,12 +252,16 @@ func SummaryToPhase(ctx context.Context, minSuccesses int64, summary arraystatus
 	}
 
 	// No chance to reach the required success numbers.
-	if totalRunning+totalSuccesses < minSuccesses {
-		logger.Infof(ctx, "Array failed early because totalRunning[%v] + totalSuccesses[%v] < minSuccesses[%v]",
-			totalRunning, totalSuccesses, minSuccesses)
+	if totalRunning+totalSuccesses+totalWaitingForResources < minSuccesses {
+		logger.Infof(ctx, "Array failed early because totalRunning[%v] + totalSuccesses[%v] + totalWaitingForResource[%v] < minSuccesses[%v]",
```

Comment:
Array failed early because number of failures are > minimum success requirements. Snapshot: TotalRunning[%d].....

Reply:
do we have concurrency limits?

Reply:
done